code/kodekit.php

```diff
@@ -75,75 +82,81 @@ class Kodekit
      */
     final private function __construct($config = array())
     {
-        //Initialize the debug state
+        /**
+         * Setup the configuration
+         */
+
         if(isset($config['debug'])) {
-            $this->_debug = $config['debug'];
+            self::$_debug = $config['debug'];
         } else {
-            $this->_debug = false;
+            self::$_debug = false;
         }
 
-        //Initialize the debug state
         if(isset($config['cache'])) {
-            $this->_cache = $config['cache'];
+            self::$_cache = $config['cache'];
         } else {
-            $this->_cache = false;
+            self::$_cache = false;
         }
 
-        //Initialize the root path
         if(isset($config['root_path'])) {
-            $this->_root_path = $config['root_path'];
+            self::$_root_path = $config['root_path'];
         } else {
-            $this->_root_path = realpath($_SERVER['DOCUMENT_ROOT']);
+            self::$_root_path = realpath($_SERVER['DOCUMENT_ROOT']);
         }
 
-        //Initialize the base path
         if(isset($config['base_path'])) {
-            $this->_base_path = $config['base_path'];
+            self::$_base_path = $config['base_path'];
         } else {
-            $this->_base_path = $this->_root_path;
+            self::$_base_path = self::$_root_path;
         }
 
-        //Initialize the vendor path
         if(isset($config['vendor_path'])) {
-            $this->_vendor_path = $config['vendor_path'];
+            self::$_vendor_path = $config['vendor_path'];
         } else {
-            $this->_vendor_path = $this->_root_path.'/vendor';
+            self::$_vendor_path = self::$_root_path.'/vendor';
         }
 
-        //Load the legacy functions
+        /**
+         * Load the legacy functions
+         */
         require_once dirname(__FILE__) . '/legacy.php';
 
-        //Setup the loader
-        require_once dirname(__FILE__).'/class/loader.php';
-
-        if (!isset($config['class_loader'])) {
-            $config['class_loader'] = Library\ClassLoader::getInstance($config);
-        }
+        /**
+         * Load the class locator
+         */
+        require_once dirname(__FILE__) . '/class/loader/loader.php';
 
-        //Setup the factory
-        $manager = Library\ObjectManager::getInstance($config);
+        $loader = new Library\ClassLoader();
+        $loader->setDebug(self::isDebug());
+        $loader->setCache(self::isCache());
```

Comment:
Took that out for now. Need to review how cache namespace setting will work. Might get it directly from the env in a next step.